From a9cf25be5b4c777ab4bcceddd2adfa7d60a97075 Mon Sep 17 00:00:00 2001
From: "Wong, Vincent Por Yin" <vincent.por.yin.wong@intel.com>
Date: Fri, 8 Nov 2019 10:12:41 +0800
Subject: [PATCH 1/3] net: stmmac: fix missing max_combined count for EHL
 platforms

max_combined is checked before creating AF_XDP sockets. For stmmac, this
value is the lowest queue count (tx/rx) divided by 2.

Signed-off-by: Wong, Vincent Por Yin <vincent.por.yin.wong@intel.com>
Signed-off-by: Voon Weifeng <weifeng.voon@intel.com>
---
 drivers/net/ethernet/stmicro/stmmac/stmmac_pci.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/net/ethernet/stmicro/stmmac/stmmac_pci.c b/drivers/net/ethernet/stmicro/stmmac/stmmac_pci.c
index c6b2d7ea73fa..f2891f168ce8 100644
--- a/drivers/net/ethernet/stmicro/stmmac/stmmac_pci.c
+++ b/drivers/net/ethernet/stmicro/stmmac/stmmac_pci.c
@@ -291,6 +291,8 @@ static int ehl_common_data(struct pci_dev *pdev,
 
 	plat->rx_queues_to_use = 8;
 	plat->tx_queues_to_use = 8;
+	/* Maximum TX XDP queue */
+	plat->max_combined = 4;
 
 	ret = intel_mgbe_common_data(pdev, plat);
 	if (ret)
-- 
2.20.1

